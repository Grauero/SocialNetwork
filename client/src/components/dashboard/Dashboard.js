import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { getCurrentProfile } from '../../store/actions/profileActions';
import Spinner from '../common/Spinner';

class Dashboard extends Component {
  componentDidMount() {
    this.props.getCurrentProfile();
  }

  render() {
    const { user } = this.props.auth;
    const { profile, loading } = this.props.profile;
    let dashboardContent;

    if (profile === null || loading) {
      dashboardContent = <Spinner />;
    } else {
      // check if user have profile
      if (Object.keys(profile).length > 0) {
        dashboardContent = '';
      } else {
        dashboardContent = (
          <div>
            <p className="lead text-muted">Welcome {user.name}</p>
            <p>You have not created profile</p>
            <Link to="/create-profile" className="btn btn-lg btn-info">
              Create profile
            </Link>
          </div>
        );
      }
    }

    return (
      <div className="dashboard">
        <div className="container">
          <div className="row">
            <div className="col-md-12">
              <h1 className="display-4">Dashboard</h1>
              {dashboardContent}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Dashboard.propTypes = {
  getCurrentProfile: propTypes.func.isRequired,
  profile: propTypes.instanceOf(Object).isRequired,
  auth: propTypes.instanceOf(Object).isRequired
};

const mapStateToProps = state => ({
  profile: state.profile,
  auth: state.auth
});

export default connect(mapStateToProps, { getCurrentProfile })(Dashboard);
