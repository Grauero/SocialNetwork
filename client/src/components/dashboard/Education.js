import React from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import Moment from 'react-moment';

import { deleteEducation } from '../../store/actions/profileActions';

const Education = ({ education, deleteEducation }) => {
  const onDeleteClick = id => deleteEducation(id);

  const educations = education.map(edu => (
    <tr key={edu._id}>
      <td>{edu.school}</td>
      <td>{edu.degree}</td>
      <td>
        <Moment format="YYYY/MM/DD">{edu.from}</Moment> -
        {edu.to === null ? (
          ' Now'
        ) : (
          <Moment format="YYYY/MM/DD">{edu.to}</Moment>
        )}
      </td>
      <td>
        <button
          type="button"
          onClick={() => onDeleteClick(edu._id)}
          className="btn btn-danger"
        >
          Delete
        </button>
      </td>
    </tr>
  ));

  return (
    <div>
      <h4 className="mb-4">Education Credentials</h4>
      <table className="table">
        <thead>
          <tr>
            <th>School</th>
            <th>Degree</th>
            <th>Years</th>
            <th />
          </tr>
          {educations}
        </thead>
      </table>
    </div>
  );
};

Education.propTypes = {
  education: propTypes.instanceOf(Object).isRequired,
  deleteEducation: propTypes.func.isRequired
};

export { Education };
export default connect(
  null,
  { deleteEducation }
)(Education);
