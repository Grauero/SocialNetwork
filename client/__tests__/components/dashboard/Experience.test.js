import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Experience } from '../../../src/components/dashboard/Experience';

const _id = 'testId';
const props = {
  experience: [{ _id }],
  deleteExperience: jest.fn()
};

const component = mount(<Experience {...props} />);

it('deletes selected education by pressing button', () => {
  component.find('button').simulate('click');

  expect(props.deleteExperience).toHaveBeenCalled();
  expect(props.deleteExperience).toHaveBeenCalledTimes(1);
  expect(props.deleteExperience).toHaveBeenCalledWith(_id);
});

it('matches snapshot', () => {
  const testProps = Object.assign({}, props, { experience: [] });
  const component = mount(<Experience {...testProps} />);

  expect(toJSON(component)).toMatchSnapshot();
});
