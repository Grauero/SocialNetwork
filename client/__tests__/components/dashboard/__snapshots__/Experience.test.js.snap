// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Experience
  deleteExperience={[Function]}
  experience={Array []}
>
  <div>
    <h4
      className="mb-4"
    >
      Experience Credentials
    </h4>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Title
          </th>
          <th>
            Years
          </th>
          <th />
        </tr>
      </thead>
    </table>
  </div>
</Experience>
`;
